realm/src/androidTest/java/io/realm/RealmQueryTest.java

```diff
@@ -459,4 +473,64 @@ public void testGeorgian() {
             assertEquals(sorted[i], stringOnlies2.get(i).getChars());
         }
     }
+
+    public void testQueryNullStringNotNullableField() {
+        try {
+            RealmResults<NullTypes> nullTypeses = testRealm.where(NullTypes.class).equalTo(NullTypes.FIELD_STRING_NOT_NULL,
+                    (String)null).findAll();
+            fail();
+        }
+        catch (IllegalArgumentException ignore) {
+        }
+    }
+
+    public void testQueryNullStringsNotNullableFieldIsNull() {
```

Comment:
I tried to create methods with names explaining what it tested. But I agree: the name is really ugly. Adding comments to unit tests is good idea!